tooling/nargo_cli/tests/stdlib-props.rs

```diff
@@ -0,0 +1,263 @@
+                if variable_length { (0..=max_len).boxed() } else { Just(max_len).boxed() };
+            // The actual input length can be up to the maximum.
+            let strategy = length_strategy
+                .prop_flat_map(|len| prop::collection::vec(any::<u8>(), len))
+                .prop_map(move |mut msg| {
+                    // The output is the hash of the data as it is.
+                    let output = hash(&msg);
+
+                    // The input has to be padded to the maximum length.
+                    let msg_size = msg.len();
+                    msg.resize(max_len, 0u8);
+
+                    let mut inputs = vec![("input", bytes_input(&msg))];
+
+                    // Omit the `message_size` if the hash function doesn't support it.
+                    if variable_length {
+                        inputs.push((
+                            "message_size",
+                            InputValue::Field(FieldElement::from(msg_size)),
+                        ));
+                    }
+
+                    SnippetInputOutput::new(inputs, bytes_input(&output)).with_description(format!(
+                        "force_brillig = {force_brillig}, max_len = {max_len}"
+                    ))
+                })
+                .boxed();
+
+            run_snippet_proptest(source.clone(), force_brillig, strategy);
+        }
+    }
+}
+
+/// This is just a simple test to check that property testing works.
+#[test]
+fn test_basic() {
+    let program = "fn main(init: u32) -> pub u32 {
+        let mut x = init;
+        for i in 0 .. 6 {
+            x += i;
+        }
+        x
+    }";
+
+    let strategy = any::<u32>()
+        .prop_map(|init| {
+            let init = init / 2;
+            SnippetInputOutput::new(
+                vec![("init", InputValue::Field(init.into()))],
+                InputValue::Field((init + 15).into()),
+            )
+        })
+        .boxed();
+
+    run_snippet_proptest(program.to_string(), false, strategy);
+}
+
+#[test]
+fn test_keccak256() {
+    run_hash_proptest(
+        // XXX: Currently it fails with inputs >= 135 bytes
+        &[0, 1, 100, 134],
+        true,
+        |max_len| {
+            format!(
+                "fn main(input: [u8; {max_len}], message_size: u32) -> pub [u8; 32] {{
+                    std::hash::keccak256(input, message_size)
+                }}"
+            )
+        },
+        |data| sha3::Keccak256::digest(data).try_into().unwrap(),
+    );
+}
+
+#[test]
+#[should_panic] // Remove once fixed
+fn test_keccak256_over_135() {
+    run_hash_proptest(
+        &[135, 150],
+        true,
+        |max_len| {
+            format!(
+                "fn main(input: [u8; {max_len}], message_size: u32) -> pub [u8; 32] {{
+                    std::hash::keccak256(input, message_size)
+                }}"
+            )
+        },
+        |data| sha3::Keccak256::digest(data).try_into().unwrap(),
+    );
+}
+
+#[test]
+fn test_sha256() {
```

Comment:
I think this test naming can be improved, can we rename these to `fuzz_sha256_equivalence`, etc.?

Reply:
Renamed